src/actions/posts.js

```diff
@@ -759,6 +760,28 @@ export function getPostsAround(channelId, postId, perPage = Posts.POST_CHUNK_SIZ
     };
 }
 
+// getThreadsForPosts is intended for an array of posts that have been batched
+// (see the actions/websocket_actions/handleNewPostEvents function in the webapp)
+export function getThreadsForPosts(posts, dispatch, getState) {
```

Comment:
@enahum I was going off of the function just below it (`getProfilesAndStatusesForPosts`) since they're both called in the same context (see: https://github.com/mattermost/mattermost-webapp/pull/3449). Does that change your suggestion?

Reply:
I would change it from `getThreadsForPosts(posts, myDispatch, myGetState);` to `myDispatch(getThreadsForPosts(posts));

Reply:
@enahum Would you replace `return Promise.all(promises)` with...?

Reply:
Changes made, thanks for the improvements @enahum !

Reply:
is ok to leave the promise.all for now as we don't really use the result of it anywhere, but for the future if you need the results you can go with an `allSettle` kind of Promise resolution